linux_os/guide/system/software/integrity/fips/grub2_enable_fips_mode/ansible/shared.yml

```diff
@@ -0,0 +1,113 @@
+# platform = multi_platform_rhel, Red Hat Enterprise Linux 7
+# reboot = true
+# strategy = enable
+# complexity = medium
+# disruption = high
+
+# Check to see if FIPS is enabled before doing anything
+
+- name: Determine if FIPS is enabled
+  shell: sysctl crypto.fips_enabled | tail -c 2
+  register: fips_state
+  changed_when: false
+  tags:
+  - always
+    @ANSIBLE_TAGS@
+
+# If system is not in FIPS mode, packages will be installed, initramfs rebuilt, prelinking disabled, grub defaults updated, kernel parameters updated, and grub rebuilt. Note the system will need to be rebooted for these changes to take into full effect.
+
+-   block:
+
+  # Install dracut-fips and dracut-fips-aesni packages
+
+  - name: Ensure dracut-fips and dracut-fips-aesni are installed
+    package:
+      name: '{{item}}'
+      state: present
+    with_items:
+    - dracut-fips
+    - dracut-fips-aesni  
```

Comment:
AES-NI is not guaranteed on all processors and environments, so this should be dropped.